addons/website_event_sale/models/event_event.py

```diff
@@ -0,0 +1,17 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models
+
+
+class Event(models.Model):
+    _inherit = 'event.event'
+
+    def _get_visitor_events(self, current_visitor):
+        events = super()._get_visitor_events(current_visitor)
+        registration_not_payed = events.registration_ids.filtered_domain([
+            ('visitor_id', '=', current_visitor.id),
+            ('payment_status', '=', 'to_pay'),
+        ])
```

Comment:
Can a logged user have different visitors? If so, we could miss some unpaid registrations here I think? (not the same visitor yet the same partner -> miss those unpaid registrations?)

Could it happen?

Reply:
The visitor is linked to the partner and a partner can have multiple visitors. So technically the user linked to a partner could also have multiple visitors indeed. But after some try, it seems pretty hard to do because once there is a partner linked to it, it seems we re-use the previous visitor.
Since the registrations are filtered with the visitor or the partner I think it should be fine to take also into account the partner, just to be safe.